spec/unit/rest/client_spec.rb

```diff
@@ -2,8 +2,61 @@
 
 require 'puppet/rest/client'
 require 'puppet/rest/ssl_context'
+require 'puppet_spec/validators'
+require 'puppet_spec/ssl'
 
 describe Puppet::Rest::Client do
+  # Follows closely with spec/unit/network/http/connection_spec's
+  # 'ssl verifier' shared context
+  shared_context 'connection error handling' do
```

Comment:
Can we use `shared_examples` here instead to make it clearer that we are sharing test cases as well as setup code?